pylint/checkers/base/basic_error_checker.py

```diff
@@ -78,6 +78,38 @@ def _has_abstract_methods(node: nodes.ClassDef) -> bool:
     return len(utils.unimplemented_abstract_methods(node)) > 0
 
 
+def _new_correctly_implemented(node: nodes.ClassDef) -> bool:
+    """Check if node implements `__new__`.
+
+    If `__new__` is implemented, check if it calls `super().__new__(cls)`.
+    """
+    if "__new__" not in node.locals:
+        return False
+
+    new = next(node.igetattr("__new__"))
+    if not isinstance(new, astroid.UnboundMethod):
+        return False
+
+    calls = new.nodes_of_class(
+        nodes.Call, skip_klass=(nodes.FunctionDef, nodes.ClassDef)
+    )
+    for call in calls:
+        if not isinstance(call.func, nodes.Attribute):
+            continue
+        if call.func.attrname != "__new__":
+            continue
```

Comment:
@Pierre-Sassoulas This is very fancy and a nice side effect of `codecov`.

Reply:
Is codecov not taking into account functional tests? Because if I add a breakpoint right before the `continue` in ln 100 my functional tests locally do stop, so I think it is covered.

Reply:
same with ln 104, it is covered by functional tests

Reply:
Nice tooltip ! Is this a bug and really is covered though ? Maybe it depends on the interpreter being used ? What do you have locally @clavedeluna ?

Reply:
Ah good call, testing with python3.8 it's covered, but with 3.11 it isn't, so I assume codecov is running with the latter? Not sure how to verify that in codecov settings, it's not specified in the yaml

Reply:
It feels weird that `coverage` doesn't see this. I would need to investigate this as I have never seen such things in other projects that use `coverage`.

Reply:
Locally running pytest with coverage I can see the following

```
pylint/checkers/base/basic_error_checker.py                    242     10    96%   284, 434-435, 489-496, 603-604
```

which tells me the lines codecov is complaining about (L98, L102) are indeed covered by tests.

Happy to continue looking into this but I would need to find this data in the action run, which I don't. So I'll block this PR for now.

Reply:
I never saw the edited comment. We don't install `pandas` so this indeed uncovered in the tests. `codecov` is right here.
The tests should work even without `pandas` in the environment as we don't have that as a test requirement.

Reply:
Aha! You found the issue, that's indeed the case. The only functional test that passes thru that code is the one using pandas. I'll add one to fix this. Thanks!